firedrake/adjoint_utils/function.py

```diff
@@ -225,11 +225,7 @@ def _ad_convert_riesz(self, value, options=None):
 
         V = options.get("function_space", self.function_space())
         if value == 0.:
-            # In adjoint-based differentiation, value == 0. arises only when
-            # the functional is independent on the control variable.
-            # In this case, we do not apply the Riesz map and return a zero
-            # Cofunction.
-            return Cofunction(V.dual())
+            return Function(V)
```

Comment:
this was done because I have lists of functionals now for each ensemble rank but only one set of controls. Each functional individually might be independent of one or more of the controls, but I need to add the results.